File: app/src/main/java/com/widyatama/nurseassistant/features/login/LoginActivity.java
package com.widyatama.nurseassistant.features.login;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.widget.AppCompatImageView;

import com.widyatama.core.base.BaseActivity;
import com.widyatama.core.util.CommonUtils;
import com.widyatama.core.util.SchedulerProviderUtil;
import com.widyatama.nurseassistant.R;
import com.widyatama.nurseassistant.data.local.LocalStorageHelper;
import com.widyatama.nurseassistant.features.main.MainActivity;

import io.reactivex.disposables.CompositeDisposable;

public class LoginActivity extends BaseActivity implements LoginViewContracts {

    private LoginPresenter<LoginViewContracts> presenter;

    @Override
    protected void onInitializedView(Bundle savedInstanceState) {
        presenter = new LoginPresenter<LoginViewContracts>(
                new LocalStorageHelper(this), new SchedulerProviderUtil(), new CompositeDisposable());

        presenter.onAttach(this);
        presenter.saveProfile();
        presenter.saveAllPatient();
        presenter.saveHealingPlan();
        presenter.saveJadwal();
        presenter.saveOtherNurse();
        presenter.saveRiwayatPenyakit();
        presenter.saveAccount();

        Button button = (Button) findViewById(R.id.mb_login);
        AppCompatImageView btnID = (AppCompatImageView) findViewById(R.id.iv_indonesia);
        AppCompatImageView btnEN = (AppCompatImageView) findViewById(R.id.iv_english);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                launchActivityClearAllStack(MainActivity.class);
            }
        });
        btnID.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CommonUtils.changeLanguage(LoginActivity.this, "in");
            }
        });
        btnEN.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CommonUtils.changeLanguage(LoginActivity.this, "en");
            }
        });

        launchActivityClearAllStack(MainActivity.class);
    }

    @Override
    protected void onDestroy() {
        presenter.onDetach();
        super.onDestroy();
    }

    @Override
    protected Integer setLayout() {
        return R.layout.activity_login;
    }

    @Override
    public void loginSuccess() {
        launchActivityClearAllStack(MainActivity.class);
    }

    @Override
    public void loginFailed() {
        Toast.makeText(this, "Login Failed", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/widyatama/nurseassistant/features/login/LoginActivity.kt
package com.widyatama.nurseassistant.features.login

import android.os.Bundle
import android.widget.Toast
import com.widyatama.core.base.BaseActivity
import com.widyatama.core.util.CommonUtil
import com.widyatama.nurseassistant.R
import com.widyatama.nurseassistant.features.main.MainActivity
import kotlinx.android.synthetic.main.activity_login.*
import org.koin.android.ext.android.inject

class LoginActivity : BaseActivity(), LoginViewContracts {

    val presenter: LoginPresenter<LoginViewContracts> by inject()

    override fun onInitializedView(savedInstanceState: Bundle?) {
        presenter.onAttach(this)
        presenter.saveProfile()
        presenter.saveAllPatient()
        presenter.saveHealingPlan()
        presenter.saveJadwal()
        presenter.saveOtherNurse()
        presenter.saveRiwayatPenyakit()
        presenter.saveAccount()

        mb_login.setOnClickListener {
            presenter.login(et_username.text.toString(), et_password.text.toString())
        }

        iv_indonesia.setOnClickListener {
            CommonUtil.changeLanguage(this, "in")
            recreate()
        }

        iv_english.setOnClickListener {
            CommonUtil.changeLanguage(this, "en")
            recreate()
        }
    }

    override fun onDestroy() {
        presenter.onDetach()
        super.onDestroy()
    }

    override fun setLayout(): Int {
        return R.layout.activity_login
    }

    override fun loginSuccess() {
        launchActivityClearAllStack(MainActivity::class.java)
    }

    override fun loginFailed() {
        Toast.makeText(this, "Login Failed", Toast.LENGTH_SHORT).show()
    }
}
